src/Widgets/ListFooter.vala

```diff
@@ -62,32 +62,28 @@ namespace SwitchboardPlugUserAccounts.Widgets {
 
         private void update_ui () {
             if (get_permission ().allowed) {
-                button_add.set_sensitive (true);
-                if (selected_user != null && selected_user != get_current_user ()
-                && !is_last_admin (selected_user) && !selected_user.get_automatic_login ()) {
-                    button_remove.set_sensitive (true);
-                    button_remove.set_tooltip_text (_("Remove user account and its data"));
-                } else {
-                    button_remove.set_sensitive (false);
-                    if (selected_user != null) {
-                        button_remove.set_tooltip_text (_("You cannot remove your own user account"));
-                    } else {
-                        button_remove.set_tooltip_text ("");
-                    }
-                }
-
-                if (get_removal_list () == null || get_removal_list ().last () == null) {
-                    hide_undo_notification ();
-                }
-            } else if (selected_user == null) {
-                button_remove.set_sensitive (false);
+                button_add.sensitive = true;
             } else {
-                button_add.set_sensitive (false);
-                button_remove.set_sensitive (false);
+                button_add.sensitive = false;
+                button_remove.sensitive = false;
                 hide_undo_notification ();
+                return;
+            }
+
+            if (selected_user == null) {
+                button_remove.sensitive = false;
+                button_remove.tooltip_text = "";
+            } else if (selected_user != get_current_user () && !is_last_admin (selected_user) && !selected_user.get_automatic_login ()) {
+                button_remove.sensitive = true;
+                button_remove.tooltip_text = _("Remove user account and its data");
+            } else {
+                button_remove.sensitive = false;
+                button_remove.set_tooltip_text (_("You cannot remove your own user account"));
             }
 
-            show_all ();
+            if (get_removal_list () == null || get_removal_list ().last () == null) {
```

Comment:
Since `get_removal_list` will always initialize a list, can we remove this `get_removal_list () == null`? It's not needed here.